### PR TITLE
Fix makefile arch dependencies

### PR DESCRIPTION
**Pull request by conbon**
* fix OS_EXT for mac
* fix ARCH shell command + add x86-> amd64 branch

### CODE REVIEW

File: ffi/Makefile [hunk 1]
```diff
@@ -14,22 +14,31 @@ ifeq ($(OS),Windows_NT)
     endif
 	OS_EXT=dll
 else
-	UNAME_OS := $(shell uname -s)
-	ifeq ($(UNAME_OS),Darwin)
-		OS_FLAG=darwin
-	endif
-	ifeq ($(UNAME_OS),Linux)
-		OS_FLAG=linux
+	ifeq ($(strip $(OS_FLAG)),)
```
Comment (vtermanis) [inline ffi/Makefile]: OS_FLAG here vs OS further above?

File: ffi/Makefile [hunk 2]
```diff
@@ -14,22 +14,31 @@ ifeq ($(OS),Windows_NT)
...
+		UNAME_OS := $(shell uname -s)
+		ifeq ($(UNAME_OS),Darwin)
```
Comment (vtermanis) [inline ffi/Makefile]: Is all this logic duplicating what go-releaser GH action does (in terms of deciding how to name artefacts)

File: ffi/Makefile [hunk 3]
```diff
@@ -14,22 +14,31 @@ ifeq ($(OS),Windows_NT)
     endif
 	OS_EXT=dll
 else
```
Comment (vtermanis) [inline ffi/Makefile]: (already here) mixture of spaces & tabs

Comment (conbon) [reply ffi/Makefile]: copied in from fabrizios csharp branch. fixed

File: ffi/Makefile (hunk 1 above)
Comment (conbon) [reply ffi/Makefile]: again, copied in from the csharp branch (which includes readme updates to show OS_FLAG override from cli).
Not relevant to this PR

File: ffi/Makefile (hunk 2 above)
Comment (conbon) [reply ffi/Makefile]: Well goreleaser doesnt seem configured for all of these - it only releases the linux_x86 binary so far by the looks of it.

Comment (conbon) [reply ffi/Makefile]: https://github.com/Iotic-Labs/iotics-identity-go/issues/39

**Review by vtermanis — APPROVED** [review]

**Pull request merged**